validators.go

```diff
@@ -15,6 +15,26 @@ func Default(keyword string) string {
 	return defaults[strings.ToLower(keyword)]
 }
 
+// DefaultAll returns the default value for the given keyword, but as a slice. If
+// there is no default for the keyword, nil is returned.
+//
+// Some multi-valued settings have different defaults based on other settings, so
+// you must provide the host alias and a function to retrieve a setting
+func DefaultAll(keyword string, alias string, get func(alias, key string) string) []string {
+	if strings.ToLower(keyword) == "identityfile" && get(alias, "Protocol") == "2" {
```

Comment:
I'm confused by this function, by this comment: 

```
// Some multi-valued settings have different defaults based on other settings, so
// you must provide the host alias and a function to retrieve a setting
```

and by the fact that this logic appears but not in `Default`. I think maybe if users want this logic they can implement it themselves, alternatively let's pull this out into a different PR.

Reply:
Originally I had put this logic in GetAll (see ee013b827b53ce170bdc56632a6db6d92eaeefb4). However, I needed to wrap the config, so I need a Default and a DefaultAll. It seems convenient to make it available to people who wanted to do such a thing.

The comment sucks. But if you look at the manpage for ssh_config (OSX 10.12), you can see that the default value for identityfile differs based on which protocol the client has specified. Though, on Linux it seems that this isn't the case. If you want to remove support for legacy stuff, then I can change this to just default to protocol 2.